### PR TITLE
test(radio-button): update to help test click behavior via HTMLElement.click()

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #4212

## Summary

Update tests that cover behavior on element click to use the .callMethod("click") in addition to .click(). In the case of the `radio-button`, click triggers a boolean toggle on the checked prop.

### CODE REVIEW

File: src/components/radio-button/radio-button.e2e.ts [hunk 1]
```diff
@@ -272,6 +285,17 @@ describe("calcite-radio-button", () => {
 
     expect(focusEvent).toHaveReceivedEventTimes(1);
     expect(blurEvent).toHaveReceivedEventTimes(1);
+
+    //in addition to .click() check with .callMethod("click") as well
```
Comment (jcfranco) [inline src/components/radio-button/radio-button.e2e.ts:289]: I don't think we need to assert on focus/blur events. The click toggling `checked` should suffice.

File: src/components/radio-button/radio-button.e2e.ts [hunk 2]
```diff
@@ -194,6 +194,19 @@ describe("calcite-radio-button", () => {
 
     expect(await first.getProperty("checked")).toBe(true);
     expect(await second.getProperty("checked")).toBe(false);
+
+    //in addition to .click() check with .callMethod("click") as well
```
Comment (jcfranco) [inline src/components/radio-button/radio-button.e2e.ts:198]: Let's rephrase this to give context behind the different clicks. The code already lets us know we're using `click` and `.callMethod("click")`, but doesn't explain why.

**Review by jcfranco — CHANGES_REQUESTED** [review]

File: src/components/radio-button/radio-button.e2e.ts (hunk 2 above)
Comment (Elijbet) [reply src/components/radio-button/radio-button.e2e.ts]: See if this solves it.

**Review by jcfranco — APPROVED** [review]
💥
